presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java

```diff
@@ -186,17 +190,45 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     {
         RaptorTableHandle handle = checkType(tableHandle, RaptorTableHandle.class, "tableHandle");
         SchemaTableName tableName = new SchemaTableName(handle.getSchemaName(), handle.getTableName());
-        List<ColumnMetadata> columns = dao.getTableColumns(handle.getTableId()).stream()
-                .map(TableColumn::toColumnMetadata)
-                .filter(isSampleWeightColumn().negate())
-                .collect(toCollection(ArrayList::new));
+        List<TableColumn> columns = dao.getTableColumns(handle.getTableId());
         if (columns.isEmpty()) {
             throw new PrestoException(RAPTOR_ERROR, "Table does not have any columns: " + tableName);
         }
 
-        columns.add(hiddenColumn(SHARD_UUID_COLUMN_NAME, SHARD_UUID_COLUMN_TYPE));
-        columns.add(hiddenColumn(BUCKET_NUMBER_COLUMN_NAME, INTEGER));
-        return new ConnectorTableMetadata(tableName, columns);
+        ImmutableMap.Builder<String, Object> properties = ImmutableMap.builder();
+        TreeMap<Integer, String> bucketing = new TreeMap<>();
+        TreeMap<Integer, String> ordering = new TreeMap<>();
+        boolean foundTemporalColumn = false;
+
+        for (TableColumn column : columns) {
+            if (column.isTemporal()) {
+                checkState(!foundTemporalColumn, "Only one column can be designated a temporal column");
+                properties.put(TEMPORAL_COLUMN_PROPERTY, column.getColumnName());
+                foundTemporalColumn = true;
```

Comment:
Also, if for whatever reason there was a code bug that allowed this, the `properties.build()` call would fail due to the duplicate entry